added a non distructive method to replace text inside a run

biggosh opened the pull request:
only <w:t> <w:br> <w:cr> <w:tab> elements are affected by this new method. So no other side effects are produced

by this new method there is a simple and safe method to replace text inside a run

Commit:
added a non distructive method to replace text inside a run
only <w:t> <w:br> <w:cr> <w:tab> elements are affected by this new method. So no other side effects are produced

Commit:
replace text at paragraph level. text is searched in all paragraph text, even if it's splitted in more run <w:r> elements

biggosh commented:
I've just pushed the replace_text method I've written for paragraphs. The method searches for text over all runs in the paragraph, even if the text is splitted over more runs. It replaces all occurrences in the paragraph. The style is the one of the first run containing the prefix of old text.

Commit:
exposed the tblCaption property

Commit:
2 methods added: copy on _Row class and insert_at_tr. The second one allows to insert a row copied or new in a specific position inside table

Commit:
add remove_row_at on table class

biggosh committed:
Update README.rst

biggosh committed:
Update README.rst

scanny commented:
Hi @biggosh, did you close this to replace it with something else or something? I'd like to leave it open otherwise, so it doesn't get lost in the ones we've merged in and so on :)

biggosh commented:
Hi @scanny. I closed this pull request because I haven't received any feedback. So I understand you are not interested in features I've implemented probably because I didn't follow all the protocol you request for extensions I've implemented or because you don't like my coding style. So let me know how to proceed. You can find my work in my repository (biggosh on github). I've extender your library with the possibility to manage tables template, adding row to an existing one in a specific position, copy a row, replace text inside a cell. I'm waiting for your feedback and instruction on how to proceed. Thanks

ps. if you prefer you can contact me directly on my email address.

scanny commented:
Hi @biggosh, you're right that we would need a certain amount of protocol in order to incorporate your changes directly, and I understand that you don't have time for that right now, that's perfectly fine with us, but that doesn't mean your work is not a contribution. I'd like to leave the pull request open if it's okay with you. Often other folks need to do something similar and will pull something from what you've done. Also when we do get around to formally including these features we'll definitely study and learn from what you've done.

If in future you do want to go through the enhancement proposal process where we agree on the API and the tests, etc., we will welcome it :)

In the meantime, if you're willing, I'd like to reopen this pull so we can make it easier for folks to find.

biggosh commented:
OK for reopening.

On 2 April 2016 at 04:25, Steve Canny <notifications@github.com> wrote:

> Reopened #274 <https://github.com/python-openxml/python-docx/pull/274>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/python-openxml/python-docx/pull/274#event-612016214>
>

lieuzhenghong commented:
This is great. @scanny, what needs to be done so that we can have it in the base branch? Can I be of help?

biggosh commented:
why the modifications (like mine replace_text) are not yet been merged?

what's missing?

thx
